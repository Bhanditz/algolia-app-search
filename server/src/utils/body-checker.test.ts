import generateValidation, { StringType, NumberType } from './body-checker';

describe('body-checker', () => {
  const testValidation = (definition, object) => {
    expect(generateValidation(definition)(object)).toMatchSnapshot();
  };

  it('can detect missing fields', () => {
    const definition = {
      title: StringType.isRequired,
    };
    const object = {};
    testValidation(definition, object);
  });

  it('ignores non-required fields', () => {
    const definition = {
      title: StringType,
    };
    const object = {};
    testValidation(definition, object);
  });

  it('can detect wrong types', () => {
    const definition = {
      title: StringType,
    };
    const object = { title: 2 };
    testValidation(definition, object);
  });

  it("doesn't error on right types", () => {
    const definition = {
      title: StringType,
    };
    const object = { title: 'piano' };
    testValidation(definition, object);
  });

  it('can work with more fields', () => {
    const definition = {
      title: StringType,
      rating: NumberType,
    };
    const object = { title: 'piano', rating: 4 };
    testValidation(definition, object);
  });

  it('validates more fields', () => {
    const definition = {
      title: StringType,
      rating: StringType,
    };
    const object = { title: 'piano', rating: 4 };
    testValidation(definition, object);
  });

  it('can throw multiple errors', () => {
    const definition = {
      title: StringType,
      rating: NumberType,
    };
    const object = { title: 2, rating: 'piano' };
    testValidation(definition, object);
  });

  it('works with null objects', () => {
    const definition = {
      title: StringType,
      rating: NumberType,
    };

    testValidation(definition, null);
  });
});
