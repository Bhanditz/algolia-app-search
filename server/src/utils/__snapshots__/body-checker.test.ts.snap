// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body-checker can detect missing fields 1`] = `
Array [
  "'title' is missing",
]
`;

exports[`body-checker can detect wrong types 1`] = `
Array [
  "title: type is number, should be string",
]
`;

exports[`body-checker can throw multiple errors 1`] = `
Array [
  "title: type is number, should be string",
  "rating: type is string, should be number",
]
`;

exports[`body-checker can work with more fields 1`] = `Array []`;

exports[`body-checker doesn't error on right types 1`] = `Array []`;

exports[`body-checker ignores non-required fields 1`] = `Array []`;

exports[`body-checker validates more fields 1`] = `
Array [
  "rating: type is number, should be string",
]
`;

exports[`body-checker works with null objects 1`] = `
Array [
  "No entity is provided",
]
`;
